desc/objectives/_reconstruction.py

```diff
@@ -1,18 +1,56 @@
 """Objectives for experimental reconstruction problems."""
 
+from abc import ABC, abstractmethod
+
 import numpy as np
 
 from desc.backend import jnp
 from desc.compute import get_profiles, get_transforms, xyz2rpz, xyz2rpz_vec
 from desc.compute.utils import _compute as compute_fun
 from desc.grid import LinearGrid
-from desc.utils import Timer, dot, errorif
+from desc.utils import Timer, dot, errorif, warnif
 
 from .normalization import compute_scaling_factors
 from .objective_funs import _Objective, collect_docs
 
 
-class PointBMeasurement(_Objective):
+class _MagneticMeasurement(_Objective, ABC):
+    """Base class for magnetic diagnostics objectives.
+
+    Subclasses of this class must implement a _compute method, which should
+    compute the same values as the compute method, but _compute should accept
+    the total B field at the necessary evaluation points as its only input,
+    and so should not compute B.
+
+    Subclasses must also define a ``self._all_eval_x_rpz`` attribute in their build
+    method, which contains every R,phi,Z point at which the total B must be evaluated
+    at. These are the points at which _compute should expect the B passed in to be
+    computed at.
+    """
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        super().build(use_jit=use_jit, verbose=verbose)
+        assert hasattr(self, "_all_eval_x_rpz"), (
```

Comment:
`_all_eval_x_rpz` might be better as an abstract property/attribute itself?

Also, the variable name implies it but should document somewhere that `x` here is in R,phi,Z. And that the compute function should expect B in R,phi,Z basis as well